### PR TITLE
Laxify the examples

### PR DESCRIPTION
**Pull request by MTCam**
This change set moves the examples closer to production, makes them lazy-ready, and updates the runner so that it can optionally run examples in lazy mode.

### CODE REVIEW

**Review by matthiasdiener — APPROVED** [review]
cc #220

File: examples/run_examples.sh [hunk 1]
```diff
@@ -3,36 +3,48 @@
...
 failed_examples=""
 for example in $examples_dir/*.py
 do
-    if [[ "$example" == *"-mpi.py" ]]
+    if [[ "$example" == *"-lazy-mpi.py" ]]
     then
-        echo "Running parallel example: $example"        
+        echo "*** Running parallel lazy example (1 rank): $example"
+        mpiexec -n 1 python -m mpi4py ${example} --lazy
+        rm -rf *vtu *sqlite *pkl *-journal restart_data
```
Comment (matthiasdiener) [inline examples/run_examples.sh:19]: Maybe add *.pvtu?

Comment (MTCam) [reply examples/run_examples.sh:19]: *vtu already gets the .pvtus!  :)

Comment (matthiasdiener) [reply examples/run_examples.sh:19]: It's late, sorry 😅

Comment (MTCam) [reply examples/run_examples.sh:19]: No worries! I realize unix is challenging late at night.

**Pull request merged**